provisioning/miniprov/test/test_cdf.py

```diff
@@ -210,5 +210,5 @@ def ret_values(value: str) -> Any:
 
         store._raw_get = Mock(side_effect=ret_values)
 
-        ret = CdfGenerator(provider=store)._create_node_descriptions()
-        self.assertEqual('None P', str(ret[0].data_iface_type))
+        with self.assertRaises(MissingKey):
+            CdfGenerator(provider=store)._create_node_descriptions()
```

Comment:
I'm sorry but this is not correct. The fact is that the method name `test_iface_type_can_be_null` states explicitly what is the predicate: it is normal if iface_type is null, our application will not break, it is expected and handled. But your change assumes that the exception will be thrown.


Please have a look at my PR - I solved the same problem there: #1548
It is fine if you copy and paste the code from there, we just need to unblock your PR and get it merged. I'll resolve conflicts in my PR then.

Reply:
I agree, this does not simulate the actual scenario.  We need to specifically cat the the null KVs that are expected to be not null.

Reply:
Done